modulus/models/module.py

```diff
@@ -65,55 +106,286 @@ def debug(self):
+        if not file_name.endswith(Module._file_extension):
+            raise ValueError(
+                f"File name must end with {Module._file_extension} extension"
+            )
+
         if file_name is None:
-            file_name = self.meta.name + ".pt"
+            raise ValueError("File name must be provided to load the model")
+
+        directory = Path(file_name)
+        if not directory.is_dir():
+            raise IOError(f"Model directory {directory} not found")
+
+        if not directory.joinpath("args.json").exists():
+            raise IOError(
+                f"Model checkpoint {directory.joinpath('model.json')} not found"
+            )
 
-        file_name = Path(file_name)
-        if not file_name.parents[0].is_dir():
+        if not directory.joinpath("metadata.json").exists():
             raise IOError(
-                f"Model checkpoint parent directory {file_name.parents[0]} not found"
+                f"Model checkpoint {directory.joinpath('metadata.json')} not found"
             )
 
-        torch.save(self.state_dict(), str(file_name))
+        if not directory.joinpath("model.pt").exists():
+            raise IOError(
+                f"Model checkpoint {directory.joinpath('model.pt')} not found"
+            )
 
-    def load(self, file_name: Union[str, None] = None) -> None:
-        """Simple utility for loading the model from checkpoint
+        # Check if the checkpoint version is compatible with the current version
+        with open(directory.joinpath("metadata.json"), "r") as f:
+            metadata_info = json.load(f)
+            if metadata_info['mdlus_file_version'] != Module.__version__:
+                raise IOError(
+                    f"Model checkpoint version {metadata_info['mdlus_file_version']} is not compatible with current version {Module.__version__}"
+                )
+
+        return directory
+
+    def load(self, file_name: str) -> None:
+        """Simple utility for loading the model weights from checkpoint
 
         Parameters
         ----------
-        file_name : Union[str,None], optional
-            Checkpoint file name. When none is provide it will default to the model's
-            name set in the meta data, by default None
+        file_name : str
+            Checkpoint file name
 
         Raises
         ------
         IOError
-            If file_name provided does not exist
+            If file_name provided does not exist or is not a valid checkpoint
         """
-        if file_name is None:
-            file_name = self.meta.name + ".pt"
 
-        file_name = Path(file_name)
-        if not file_name.exists():
-            raise IOError(f"Model checkpoint {file_name} not found")
+        directory = Module._check_checkpoint(file_name)
 
-        model_dict = torch.load(file_name, map_location=self.device)
+        model_dict = torch.load(
+            directory.joinpath("model.pt"), map_location=self.device
```

Comment:
Would prefer if `map_location` was exposed in the args of this method but not sure how that affects other parts of the code that use `self.device`